gitk-git/gitk

```diff
@@ -250,8 +250,11 @@ proc parseviewargs {n arglist} {
 		set nextisval 1
 		lappend glflags $arg
 	    }
-	    "--not" - "--all" {
+	    "--not" {
 		lappend revargs $arg
+	    }
+		"--all" {
+		set allknown 0
```

Comment:
The diff does not immediately explain what the existing purpose of the `allknown` variable is. To avoid reviewer puzzlement, the commit message offers an excellent canvas to describe the reasoning behind this diff.

Also: you might want to prefix the first line of the commit message with `gitk: ` to make it easier to read the resulting commit history.

Reply:
I think I your list done.. let me know what is next.

Reply:
The first line of the commit message seems still to be too long, see https://github.com/gitgitgadget/git/pull/375/commits/5c77ec6a7dd39c33e68c9b0e178b1c37c01ffbe8 (the ... at the end of the first line, and the fact that it is broken into two lines, suggest this to me).

Further, I was still puzzled about the role of `allknown`. So I dug a little deeper and found this:

```tcl
		    # a flag argument that we don't recognize;
		    # that means we can't optimize
		    set allknown 0
```

Maybe you can copy that comment, that really helped me understand.

Finally, your commit message seems to reflect only the analysis part, not the part where it actually describes the change (something like a paragraph at the end would do wonders, e.g. starting with "So let's change the code to prevent that `--all` option from being expanded, imitating how `-<n>` is handled.").

It would likely be a good idea to check the grammar once more (start sentences with an upper-case letter, end them in a period character) and amend.

Finally send the patch to the mailing list for review. You can use [GitGitGadget](https://gitgitgadget.github.io), [submitGit](https://submitgit.herokuapp.com) or [send it manually](https://github.com/git-for-windows/git/blob/master/CONTRIBUTING.md#submit-your-patch).

Reply:
Thanks, if you are OK with this revision I can submit it?

Reply:
Of course! I am not a gatekeeper ;-) And I see that you already submitted it :-)